platform/linux-generic/test/validation/api/shmem/shmem_linux.c

```diff
@@ -226,9 +226,12 @@ int main(int argc __attribute__((unused)), char *argv[])
 	if (exeext == NULL)
 		exeext = "";
 
+	strncpy(dir_name, argv[0], PATH_MAX - 1);
+	dir_name[PATH_MAX - 1] = 0;
+	dirname(dir_name);
```

Comment:
It would be clearer to save `dirname()` return value and use that. `dirname()` path argument handling may vary by implementation.

Reply:
v3.